### PR TITLE
Handle correctly octal or hex values located in template strings

### PR DESCRIPTION
**Pull request by ypapouski**
The PR fixes these two issues:
https://github.com/surfly/it/issues/4752
https://github.com/surfly/it/issues/4815

In general the PR fixes these two corner cases with template strings:
1. tag()\`template string with octal or hex value\`
2. String.raw\`template string with octal or hex value\`

### CODE REVIEW

File: test/parser/expressions/template.ts [hunk 1]
```diff
@@ -338,7 +338,15 @@ describe('Expressions - Template', () => {
...
-    'new nestedNewOperatorFunction`1``2``3``array`'
+    'new nestedNewOperatorFunction`1``2``3``array`',
+    "tag()`'\\00a0'`;",
+    "(tag = () => {})`'\\00a0'`",
+    "(() => {})`'\\00a0'`",
+    "(function tag() { return () => {}; })()`'\\00a0'`",
+    "(function() { return () => {}; })()`'\\00a0'`",
+    "(function tag() {})`'\\00a0'`",
+    "(function() {})`'\\00a0'`",
+    'String.raw`{\rtf1adeflang1025ansiansicpg1252\\uc1`;'
```
Comment (DimaIT) [inline test/parser/expressions/template.ts:349]: Looks like the case from [the ticket](https://github.com/surfly/it/issues/4752) is still not working.
Tested on https://meriyah.github.io/meriyah/

```
String.raw`{\rtf1\adeflang1025\ansi\ansicpg1252\uc1`;

// > [1:49]: Invalid hexadecimal escape sequence
```

Comment (ypapouski) [reply test/parser/expressions/template.ts:349]: Actually none of my examples are working in this app and I believe it's because that my fix is available only in version v4.4.2. The version of `https://meriyah.github.io/meriyah/` is v4.3.4
Screenshot 2024-04-04 at 11 31 28

Comment (ypapouski) [reply test/parser/expressions/template.ts:349]: There is a specific test for this case https://github.com/surfly/meriyah/pull/29/files#diff-9da4b71cb6cc09891f2ad28992c7cf414cf7c39d26adcc67c66069f22d4e1684R4132 and two more for other main cases.

**Review by pmenichelli — APPROVED** [review]

File: test/parser/expressions/template.ts (hunk 1 above)
Comment (DimaIT) [reply test/parser/expressions/template.ts:349]: You're right, online version of the parser is outdated.

But the test cases are actually working in old versions, that caused the confusion:

```js
// string from the tests - parsed before the fix
String.raw`{\rtf1adeflang1025ansiansicpg1252\\uc1`;

// string from the ticket - only works after the fix
String.raw`{\rtf1\adeflang1025\ansi\ansicpg1252\uc1`;
```
Can we add second string to the tests?

**Review by DimaIT — APPROVED** [review]

File: test/parser/expressions/template.ts (hunk 1 above)
Comment (ypapouski) [reply test/parser/expressions/template.ts:349]: Aha, I see your confusion and it's tricky indeed! I must have added an additional '\' because otherwise a js string can't be created:
Screenshot 2024-04-04 at 18 47 16

Comment (DimaIT) [reply test/parser/expressions/template.ts:349]: Not as a regular string, but `String.raw` actually accepts it.
image
